src/firmware.c

```diff
@@ -42,6 +42,58 @@ pk11_offs *pkg11_offsentify(u8 *pkg1) {
     return NULL;
 }
 
+void patchFS(pkg2_kip1_info_t* ki) {
+    u8 kipHash[0x20];
+
+    print("Patching FS\n");
+          
+    se_calc_sha256(&kipHash, ki->kip1, ki->size);
+    se_calc_sha256(&kipHash, ki->kip1, ki->size);
+
+    //Create header
+    size_t sizeDiff = ki->kip1->sections[0].size_decomp - ki->kip1->sections[0].size_comp;
+    size_t newSize = ki->size + sizeDiff;
+    pkg2_kip1_t *moddedKip = malloc(newSize);
+    memcpy(moddedKip, ki->kip1, newSize);
+    u32 pos = 0;
+    for(int i = 0; i < KIP1_NUM_SECTIONS; i++) {
+        if(!i) {
+            //Get decomp .text segment
+            u8 *kipDecompText = blz_decompress(moddedKip->data, moddedKip->sections[i].size_comp);
+
+            kippatchset_t *pset = kippatch_find_set(kipHash, kip_patches);
+            if (!pset) {
+              print("  could not find patchset with matching hash\n");
```

Comment:
Why is this first code branch inside the loop? Can't you run this before the loop and just start `i` from 1?

Reply:
This part of code is written by Reisyukaku

Reply:
Ah, my bad... Can you fix it though? 😊

Reply:
Okey, I am building it, and I am gonna test it in my switch